docs/docs/dev_docs/contracts/compiling.md

```diff
@@ -1,22 +1,136 @@
+
+To generate them, include a `--typescript` option in the compile command with a path to the target folder for the typescript files:
+
+```
+aztec-cli compile --typescript ./src/ts/contracts ./src/noir
+```
+
+Example code generated from the [PrivateToken](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/private_token_contract/src/main.nr) contract:
+
+```ts showLineNumbers
+export class PrivateTokenContract extends ContractBase {
+  /** Creates a contract instance at the given address. */
+  public static async at(address: AztecAddress, wallet: Wallet) { ... }
+
+  /** Creates a tx to deploy a new instance of this contract. */
+  public static deploy(rpc: AztecRPC, initial_supply: FieldLike, owner: FieldLike) { ... }
+
+  /** Type-safe wrappers for the public methods exposed by the contract. */
+  public methods!: {
+    /** getBalance(owner: field) */
+    getBalance: ((owner: FieldLike) => ContractFunctionInteraction) & Pick<ContractMethod, 'selector'>;
+
+    /** mint(amount: field, owner: field) */
+    mint: ((amount: FieldLike, owner: FieldLike) => ContractFunctionInteraction) & Pick<ContractMethod, 'selector'>;
+
+    /** transfer(amount: field, sender: field, recipient: field) */
+    transfer: ((amount: FieldLike, sender: FieldLike, recipient: FieldLike) => ContractFunctionInteraction) &
+      Pick<ContractMethod, 'selector'>;
+  };
+}
+```
+
+### Noir interfaces
+
+A Noir contract can [call a function](./functions.md) in another contract via `context.call_private_function` or `context.call_public_function`. However, this requires manually assembling the function selector and manually serialising the arguments, which is not type-safe.
+
+To make this easier, the compiler can generate a contract interface struct that exposes a convenience method for each function listed in a given contract ABI. These structs are intended to be used from another contract project that calls into the current one.
+
+To generate them, include a `--interface` option in the compile command with ap ath to the target folder for the generated Noir interface files:
+
+```
+aztec-cli compile --interface ../client/src/noir ./src/noir
+```
+
+Example code generated from the [PrivateToken](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/private_token_contract/src/main.nr) contract:
+
+```rust
+impl PrivateTokenContractInterface {
+  fn at(address: Field) -> Self {
+      Self { address }
+  }
+  
+  fn mint(
+    self, context: &mut PrivateContext, amount: Field, owner: Field
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 2];
+    serialised_args[0] = amount;
+    serialised_args[1] = owner;
+
+    context.call_private_function(self.address, 0x1dc9c3c0, serialised_args)
```

Comment:
sorry if dumb question, but what are these two `0x...` values?

Reply:
Hi! :) No, not a dumb question at all! They're 'function selectors'. Much like ethereum, a function can be represented by a 4-byte hex identifier, derived as the hash of the function name and its parameter types. (There's a `compute_function_selector` function somewhere in the codebase (I might have got that function name wrong)).
These hex-values are hard-coded here at the time we auto-generate a contract's interface.
Note: we might change the function selector to be more than 4-bytes in future, to reduce the likelihood of collisions. TBD.

Reply:
Since the first person to read this doc (Dan) had this question, we should add an explanation (or a hyperlink to an explanation) of what the hex values are.

Reply:
ahh ok great thanks!  got it thank you Michael, good to know we also have function selectors here

Reply:
Added a comment with the explanation in the code.